libnixf/src/Sema/VariableLookup.cpp

```diff
@@ -52,8 +52,24 @@ void VariableLookupAnalysis::emitEnvLivenessWarning(
     if (!Def->syntax())
       continue;
     if (Def->uses().empty()) {
-      Diagnostic &D = Diags.emplace_back(Diagnostic::DK_DefinitionNotUsed,
-                                         Def->syntax()->range());
+      Diagnostic::DiagnosticKind kind;
```

Comment:
I'd like to see an IIFE for this, something like:

```suggestion
      Diagnostic::DiagnosticKind kind = [&](){
        switch(Def->source()) {
        case Definition::DS_Let:
          return Diagnostic::DK_UnusedDefLet;
        case Definition::DS_LambdaFormal:
          return Diagnostic::DK_UnusedDefFormal;
        /* ... etc ... */
        }
      }();
```